pkg/cmd/server/admin/create_signercert.go

```diff
@@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
 		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
 	}
 	if written {
-		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
+		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
```

Comment:
> On Jun 5, 2015, at 9:37 AM, David Eads <notifications@github.com> wrote:
> 
> In pkg/cmd/server/admin/create_signercert.go:
> 
> > @@ -99,9 +96,9 @@ func (o CreateSignerCertOptions) CreateSignerCert() (*crypto.CA, error) {
> >  		ca, written, err = crypto.EnsureCA(o.CertFile, o.KeyFile, o.SerialFile, o.Name)
> >  	}
> >  	if written {
> > -		fmt.Fprintf(o.Output.Get(), "Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
> > +		glog.V(3).Infof("Generated new CA for %s: cert in %s and key in %s\n", o.Name, o.CertFile, o.KeyFile)
> We have log levels for this.
> 
> I'd tend to disagree. When you explicitly running create-signer-cert, you clearly want this message. When you're delegating to this method because you ran an openshift start server, the relative importance of these messages changes. That can be handled by not having a library type method perform output, but that seems heavy in this case. It could also be handled by a filtering chain on top of an io.Writer which seems far more reasonable in this case.
> 
The problem is putting human messages in code reused in different contexts.  The general solution to that is to print messages above the shared code and make sure you respond correctly.  Filtering output is focusing on structuring return values out of unstructured content, which usually is a mistake
> —
> Reply to this email directly or view it on GitHub.
>